src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -321,14 +341,14 @@ export class RunParametersAction extends TooltipFromLabelAction {
 		let inputParameters = new Map<string, string>();
 		let uriParams = new URLSearchParams();
 		// Store new parameter values to map based off defaultParameters
-		if (defaultParameters.size === 0) {
+		if (defaultParameters.size === 0 && !emptyParameterCell) {
```

Comment:
Why are we even getting down here if we have an empty parameters cell? As mentioned above I think we should just exit early in that case.

Reply:
I do exit the notification and after seeing if a cell.parameter is found. However, this only exits the loop of the for loop and not the entire action and therefore still needs to make this condition to make sure we don't do anything further for an empty parameter cell.

Reply:
The return is in a forEach so it'll only exit that foreach.

Just convert the forEach function into an actual for loop instead and then you can return and not bother with these checks.